assets/js/edit.js

```diff
@@ -144,6 +156,19 @@ class Edit extends Component {
 			</BlockControls>
 		);
 
+		const { getCurrentPost } = select('core/editor');
+		const postDetails = getCurrentPost();
```

Comment:
It would be good to subscribe to the post data so the block is updated when the post title is edited https://gutenberg.10up.com/guides/data-api/#examples

Reply:
@peterwilsoncc Could you please add some example snippet here? I was trying to use data-api example from above link but getting error called Invalid hook call since it's a class component. Tried in componentDidMount lifecycle hook as well got same error. Thanks in advance :)